### ISSUE TITLE
Provide a stricter version of no-cond-assign

### ISSUE THREAD

**Issue by hegemonic** [eslint/eslint#1444]
The `no-cond-assign` rule does not catch assignments that are enclosed in parentheses, such as:

``` js
if (foo && (bar = baz)) {
  // ...
}
```

It looks like this was an intentional decision (see #505), and as the default behavior, this seems fine. But I'd like a rule to catch _any_ assignment in a conditional.

Are you open to adding an argument to `no-cond-assign`? If so, I can submit a pull request. Perhaps the argument could be `parenthesised` for the current (default) behavior or `never` to prohibit any assignment in a conditional.

Either way, I think it would be helpful to spell out the current behavior more clearly in the `no-cond-assign` docs. I can submit a pull request for that, too.
